src/main/java/seedu/address/model/module/DetailedModuleList.java

```diff
@@ -0,0 +1,66 @@
+package seedu.address.model.module;
+
+import java.util.ArrayList;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.module.exceptions.ModuleNotFoundException;
+
+/**
+ * List of Modules.
+ */
+public class DetailedModuleList {
+    private ArrayList<Module> modules;
+
+    public DetailedModuleList() {
+        this.modules = new ArrayList<>();
+    }
+
+    public void addModule(Module module) {
+        this.modules.add(module);
+    }
+
+    /**
+     * Finds a module using the academic year and module code, and returns the module.
+     *
+     * @param acadYear academic year
+     * @param code of the module to find
+     * @return module found
+     */
+    public Module findModule(AcadYear acadYear, ModuleCode code) {
+        for (int i = 0; i < modules.size(); i++) {
+            Module module = modules.get(i);
+            if (module.getAcadYear().equals(acadYear) && module.getModuleCode().equals(code)) {
+                return modules.get(i);
```

Comment:
Consider reusing the `module` variable as declared on line 32 for increased clarity.